client/trino-client/src/main/java/io/trino/client/auth/kerberos/DelegatedUnconstrainedContextProvider.java

```diff
@@ -55,4 +57,16 @@ public void refresh()
             }
         }
     }
+
+    private static Subject current()
+    {
+        try {
+            Method current = Subject.class.getDeclaredMethod("current");
+            return (Subject) current.invoke(null);
+        }
+        catch (NoSuchMethodException | IllegalAccessException | InvocationTargetException ignored) {
+            // Fallback to pre-Java 17 method
+            return Subject.getSubject(AccessController.getContext());
+        }
+    }
```

Comment:
Maybe it's time to bump the Java version from the ancient 8 to something less ancient in trino-client?

Reply:
17 is not an option yet.